addons/web/static/src/js/views/list/list_renderer.js

```diff
@@ -745,6 +745,9 @@ var ListRenderer = BasicRenderer.extend({
             return $th;
         }
         var description = node.attrs.string || field.string;
+        if (node.attrs.class && node.attrs.class.indexOf('oe_read_only') !== -1) {
+            $th.addClass('oe_read_only');
+        }
```

Comment:
this commit looks like a framework stuff as it will apply to all editable list, did you discuss it wit anyone of the framework team? i think we should move this commit in another framework PR

Reply:
@sle-odoo We already have a code where if ```oe_read_only``` class added then that cell will be hidden but thead was still taking space that's why we have fixed this framework issue, we have added test for the same, you can request framework team guy to review it(we have separate commit for this)

Reply:
alright, but my point is please remove this commit from this pr and create a new one, this way i can go forward with the merge of the task

also i suppose that if one line has a readonly field and the next one not there will be a display issue if you hide it but i may be wrong

Reply:
new PR created for this commit